app/views/comfy/admin/api_actions/_form.html.haml

```diff
@@ -1,125 +1,125 @@
-    } else {
-        setFields()
-    }
-
-    function setFields() {
-        let str = JSON.stringify(editor.get())
-        $('#' + fieldId).val(str)
-    }
-
-    const updatedJson = editor.get()
-  }
-
+.card.form-container.font-size-sm.position-relative.p-3.mb-4{id: "api_action_form_#{type}_#{index}", style: "font-size: 13px; border-color: #3883fa"}
+  .position-absolute{style: "top: 6px; right: 6px"}
+    %a.text-danger{style: "cursor: pointer;", onclick: "removeForm('api_action_form_#{type}_#{index}', 'api_action_form_destroy_field_#{type}_#{index}')"}
+      %i.fas.fa-times
+  .row
+    .form-group.col-12
+      = label_tag "Action"
+      = select_tag "api_namespace[#{type}_attributes][#{index}][action_type]", options_for_select(ApiAction.action_types.keys, resource.action_type), { class: "form-control form-control-sm", id: "action_type_field_#{type}_#{index}", onchange: "toggleApiActionFields('#{index}', '#{type}');"}
+      - if resource.action_type == 'send_email' || resource.action_type == 'send_web_request'
+        .form-group.my-2
+          .d-block
+            %strong Top-Level Model Attributes: 
+            .d-block
+              = @api_namespace&.properties&.keys
+        - last_instance = @api_namespace&.api_resources&.last
+        - if last_instance
+          .form-group
+            .d-block
+              %strong Last API Resource Instance Attributes: 
+              .d-block
+                = last_instance.properties.keys
+          .d-block
+            %small
+              Dynamic:
+              %code
+                \\#{api_resource.properties['your_custom_attribute_here']} 
+
+    .send_email.col-12{style: "#{'display:none' unless resource.action_type == 'send_email'}", id: "send_email_fields_#{type}_#{index}"}
+      .form-group
+        -# this is how we allow the user to unselect the selected checkbox. Using jquery and this hidden field to set the value to false if unchecked
+        = hidden_field_tag "api_namespace[#{type}_attributes][#{index}][include_api_resource_data]", false
+        = label_tag 'Include Api Resource Data'
+        = check_box_tag "api_namespace[#{type}_attributes][#{index}][include_api_resource_data]", resource.include_api_resource_data , resource.include_api_resource_data, onClick: "$(this).attr('value', this.checked ? 1 : 0)"
+      .form-group
+        = label_tag 'Email Address'
+        = text_field_tag "api_namespace[#{type}_attributes][#{index}][email]", resource.email, { class: "form-control form-control-sm" }
+
+      .form-group
+        = label_tag 'Subject'
+        =  text_field_tag "api_namespace[#{type}_attributes][#{index}][email_subject]", resource.email_subject, { class: "form-control form-control-sm" }
+
+      .form-group
+        = label_tag 'Custom Message'
+        = rich_text_area_tag "api_namespace[#{type}_attributes][#{index}][custom_message]", resource.custom_message, { class: "form-control form-control-sm", style: "height: auto;" }

```

Comment:
@Ayon95 come to think of it, I don't think we should merge this PR because this diff is far too big for just this change. Can you double check why this is the case? It's not typical to make a small change like this and get a huge diff like this.